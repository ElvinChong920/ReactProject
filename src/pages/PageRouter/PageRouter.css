.Link-container {
    display: flex;
    justify-content: space-evenly;
    margin-top: 8px;
    margin-bottom: 16px;
}

.Link {
    background-color: lightseagreen;
    text-decoration: none;
    padding: 8px 16px;
    border-radius: 4px;
    color: white;
    width: 120px;
}
