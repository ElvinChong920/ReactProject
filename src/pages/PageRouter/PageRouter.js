import React, { Component, Fragment } from 'react';
import Link from 'redux-first-router-link';
import Welcome from '../Welcome/Welcome';
import TodoList from '../TodoList/TodoList';
import NotFound from '../NotFound/NotFound';
import Pokemon from '../Pokemon/Pokemon';
import User from '../User/User';
import UserDetail from '../User/UserDetail/UserDetail';
import { WELCOME, TODOLIST, USER, USERDETAIL, POKEMON } from '../../store/Route/routeNames';
import { connect } from 'react-redux';
import { NOT_FOUND } from 'redux-first-router';
import './PageRouter.css';

class PageRouter extends Component {

    Pages = {
        [WELCOME]: <Welcome />,
        [TODOLIST]: <TodoList />,
        [USER]: <User />,
        [USERDETAIL]: <UserDetail />,
        [NOT_FOUND]: <NotFound />,
        [POKEMON]: <Pokemon />
    };

    render () {
        const { location } = this.props;
        console.log(location);
        return (
            <Fragment>
                <div className='Link-container'>
                    <Link className='Link' to={{type: WELCOME}}>Welcome</Link>
                    <Link className='Link' to={{type: TODOLIST}}>To-do List</Link>
                    <Link className='Link' to={{type: USER}}>User</Link>
                    <Link className='Link' to={{type: POKEMON}}>Pokedex</Link>
                </div>
                {/* {this.Pages[location.type]} */}
            </Fragment>
        );
    }
}

const mapStateToProps = state => {
    return { location: state.location };
};

export default connect(mapStateToProps)(PageRouter);
