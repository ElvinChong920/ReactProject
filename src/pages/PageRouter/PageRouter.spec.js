import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import PageRouter from './PageRouter';

describe('Page Router', ()=>{
    const store = configureStore([thunk])();

    it('should render without crashing',()=>{
        // const props = {
        //     location:{
        //         type: 'USER'
        //     }
        // };
        const wrapper = shallow(<PageRouter store={store} location={{type:'USER'}}/>).dive();
        expect(wrapper).toMatchSnapshot();
    });
});
