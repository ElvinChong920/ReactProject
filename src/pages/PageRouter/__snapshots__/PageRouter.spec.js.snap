// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Router should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageRouter
    dispatch={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="Link-container"
      >
        <Connect(Link)
          className="Link"
          to={
            Object {
              "type": "WELCOME",
            }
          }
        >
          Welcome
        </Connect(Link)>
        <Connect(Link)
          className="Link"
          to={
            Object {
              "type": "TODOLIST",
            }
          }
        >
          To-do List
        </Connect(Link)>
        <Connect(Link)
          className="Link"
          to={
            Object {
              "type": "USER",
            }
          }
        >
          User
        </Connect(Link)>
        <Connect(Link)
          className="Link"
          to={
            Object {
              "type": "POKEMON",
            }
          }
        >
          Pokedex
        </Connect(Link)>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Link)
            className="Link"
            to={
              Object {
                "type": "WELCOME",
              }
            }
          >
            Welcome
          </Connect(Link)>,
          <Connect(Link)
            className="Link"
            to={
              Object {
                "type": "TODOLIST",
              }
            }
          >
            To-do List
          </Connect(Link)>,
          <Connect(Link)
            className="Link"
            to={
              Object {
                "type": "USER",
              }
            }
          >
            User
          </Connect(Link)>,
          <Connect(Link)
            className="Link"
            to={
              Object {
                "type": "POKEMON",
              }
            }
          >
            Pokedex
          </Connect(Link)>,
        ],
        "className": "Link-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Welcome",
            "className": "Link",
            "to": Object {
              "type": "WELCOME",
            },
          },
          "ref": null,
          "rendered": "Welcome",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "To-do List",
            "className": "Link",
            "to": Object {
              "type": "TODOLIST",
            },
          },
          "ref": null,
          "rendered": "To-do List",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "User",
            "className": "Link",
            "to": Object {
              "type": "USER",
            },
          },
          "ref": null,
          "rendered": "User",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Pokedex",
            "className": "Link",
            "to": Object {
              "type": "POKEMON",
            },
          },
          "ref": null,
          "rendered": "Pokedex",
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="Link-container"
        >
          <Connect(Link)
            className="Link"
            to={
              Object {
                "type": "WELCOME",
              }
            }
          >
            Welcome
          </Connect(Link)>
          <Connect(Link)
            className="Link"
            to={
              Object {
                "type": "TODOLIST",
              }
            }
          >
            To-do List
          </Connect(Link)>
          <Connect(Link)
            className="Link"
            to={
              Object {
                "type": "USER",
              }
            }
          >
            User
          </Connect(Link)>
          <Connect(Link)
            className="Link"
            to={
              Object {
                "type": "POKEMON",
              }
            }
          >
            Pokedex
          </Connect(Link)>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(Link)
              className="Link"
              to={
                Object {
                  "type": "WELCOME",
                }
              }
            >
              Welcome
            </Connect(Link)>,
            <Connect(Link)
              className="Link"
              to={
                Object {
                  "type": "TODOLIST",
                }
              }
            >
              To-do List
            </Connect(Link)>,
            <Connect(Link)
              className="Link"
              to={
                Object {
                  "type": "USER",
                }
              }
            >
              User
            </Connect(Link)>,
            <Connect(Link)
              className="Link"
              to={
                Object {
                  "type": "POKEMON",
                }
              }
            >
              Pokedex
            </Connect(Link)>,
          ],
          "className": "Link-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Welcome",
              "className": "Link",
              "to": Object {
                "type": "WELCOME",
              },
            },
            "ref": null,
            "rendered": "Welcome",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "To-do List",
              "className": "Link",
              "to": Object {
                "type": "TODOLIST",
              },
            },
            "ref": null,
            "rendered": "To-do List",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "User",
              "className": "Link",
              "to": Object {
                "type": "USER",
              },
            },
            "ref": null,
            "rendered": "User",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Pokedex",
              "className": "Link",
              "to": Object {
                "type": "POKEMON",
              },
            },
            "ref": null,
            "rendered": "Pokedex",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
